app/src/main/java/com/nmakademija/nmaakademija/fragment/TimeUntilSessionFragment.java

```diff
@@ -38,43 +41,66 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
+            setData();
     }
 
-    private void getData() {
+    private void setData(){
+        View view = getView();
+        if (view != null) {
+            TextView timeUntilSessionTextTV = (TextView) view.findViewById(R.id.timeUntilSessionText);
 
-        API.nmaService.getTimeTillSession().enqueue(new Callback<TimeTillSession>() {
-            @Override
-            public void onResponse(Call<TimeTillSession> call, Response<TimeTillSession> response) {
-                View view = getView();
-                if (view != null) {
-                    TimeTillSession timeTillSession = response.body();
-                    timeUntilSession = new TimeUntilSession(timeTillSession.getStartTime(), timeTillSession.getEndTime());
-
-                    TextView timeUntilSessionTextTV = (TextView) view.findViewById(R.id.timeUntilSessionText);
+            timeUntilSessionTextTV.setText(timeUntilSession.isSession() ? getString(R.string.timer_session_end) : getString(R.string.timer_session_start));
 
-                    timeUntilSessionTextTV.setText(timeUntilSession.isSession() ? getString(R.string.timer_session_end) : getString(R.string.timer_session_start));
+            final TextView timeUntilSessionTV;
+            timeUntilSessionTV = (TextView) getView().findViewById(R.id.timeUntilSession);
+            countDownTimer = new CountDownTimer(( timeUntilSession.isSession() ? timeUntilSession.getEndTime() : timeUntilSession.getStartTime() ).getTime() - new Date().getTime(), 1000) {
+                public void onTick(long millisUntilFinished) {
+                    if (isVisible()) {
+                        timeUntilSessionTV.setText(timeUntilSession.returnTime(getContext()));
+                    }
+                }
 
-                    countDownTimer = new CountDownTimer(timeTillSession.getEndTime().getTime(), 1000) {
-                        public void onTick(long millisUntilFinished) {
-                            if (isVisible()) {
-                                timeUntilSessionTV.setText(timeUntilSession.returnTime(getContext()));
-                            }
-                        }
+                @Override
+                public void onFinish() {
+                    setData();
+                }
+            };
 
-                        @Override
-                        public void onFinish() {
+            countDownTimer.start();
+        }
+    }
 
-                        }
-                    };
+    private void getData() {
 
-                    countDownTimer.start();
-                }
+        API.nmaService.getTimeTillSession().enqueue(new Callback<TimeTillSession>() {
+            @Override
+            public void onResponse(Call<TimeTillSession> call, Response<TimeTillSession> response) {
+                TimeTillSession timeTillSession = response.body();
+                timeUntilSession = new TimeUntilSession(timeTillSession.getStartTime(), timeTillSession.getEndTime());
+                setData();

```

Comment:
Interesting choice to save `TimeUntilSession` instead of `TimeTillSession` data object. But let's leave as it is.